### PR TITLE
feat: Added `devenv devspace` proxy command

### PR DESCRIPTION
**Pull request by plisy**
## What this PR does / why we need it
Adding `devspace` command as proxy for using `devspace` with `dev-environment` context.

**JIRA ID**: XX-XX

### CODE REVIEW

File: cmd/devenv/devspace/devspace.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+			return o.Run(c.Context)
+		},
+	}
+}
+
+func (o *Options) Run(ctx context.Context) error {
+	_, err := exec.LookPath("devspace")
+	if err != nil {
+		return errors.Wrap(err, "failed to find devspace CLI. run `orc setup` to install it")
```
Comment (jaredallard) [inline cmd/devenv/devspace/devspace.go:62]: orc isn't open-source, can't mention it here. Use the downloader framework instead like we do w/ kind, loft, and other deps.

File: cmd/devenv/devspace/devspace.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	if !contextSet {
+		o.args = append(o.args, "--kube-context", "dev-environment")
+	}
+	// We are specifying namespaces in devspace.yaml and kube context here. No need to frighten people with useless warnings.
+	if !noWarn && !contextSet {
+		o.args = append(o.args, "--no-warn")
+	}
+
+	//nolint:gosec // Why: This is a pass through command to devspace, we want to pass through all arguments.
+	cmd := exec.CommandContext(ctx, "devspace", o.args...)
```
Comment (jaredallard) [inline cmd/devenv/devspace/devspace.go:105]: use `cmdutil.RunKubernetesCommand` here instead, that does the same as `--kube-context` but ensure that it only has access to the correct KUBECONFIG

File: cmd/devenv/devspace/devspace.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+	cmd.Stdout = os.Stdout
+	cmd.Stdin = os.Stdin
+	cmd.Stderr = os.Stderr
+
+	return cmd.Run()
```
Comment (jaredallard) [inline cmd/devenv/devspace/devspace.go:111]: decorate this error, sometimes it's not clear when it says `failed to run: exit code 1`, use `errors.Wrap(cmd.Run(), "failed to wrap devspace")` or something like that.

**Review by jaredallard — CHANGES_REQUESTED** [review]

**Comment by jaredallard** [comment]
I'm not really sure how I feel about adding a devspace specific command to this when we've specifically avoided being specific to implementations in all of these commands. I'll need to think about this for a bit.

**Comment by plisy** [comment]
The idea was to follow `kubectl` as example. This way we can experiment with using devspace, without having to deal with context issues and integrating it fully into devenv commands.

I plan to integrate it fully, but that's a longer effort. In the meantime we can have people trying it out and reporting issues.

**Comment by jaredallard** [comment]
Yeah, but once we commit to `devenv devspace` as a signature we need to then make a breaking change to remove it later which I don't really want to do any time soon.

Can we rename this to `devenv apps dev` and just stub the `apps` part? I have a larger PR that renamed `deploy-app`, `update-app` and `delete-app` to `apps` counterparts.

Then, in that command, just check if there exists a `devspace.yaml` and run devspace if so. I really don't want to tie this to specific solutions. Hence why `devenv tunnel` is a thing vs `devenv localizer`

**Pull request closed without merging**